pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/worker/ConnectorsManager.java

```diff
@@ -72,6 +108,28 @@ public Path getSinkArchive(String sinkType) {
 
     public void reloadConnectors(WorkerConfig workerConfig) throws IOException {
         connectors = ConnectorUtils
-                .searchForConnectors(workerConfig.getConnectorsDirectory(), workerConfig.getNarExtractionDirectory());
+                .searchForConnectors(workerConfig.getConnectorsDirectory(), workerConfig.getNarExtractionDirectory(),
+                        workerConfig.getConnectorsCatalogueUrl());
+    }
+
+    private File loadArchiveFromDownloadPath(String archiveName, Connector connector)
+            throws IOException {
+        try {
+            File localArchive;
+            if (connector.getDownloadPath().startsWith(Utils.HTTP)) {
```

Comment:
we should log here something to make it clear that we are downloading from the Internet the connector and also the directory